lib/avalon/ffprobe.rb

```diff
@@ -0,0 +1,59 @@
+# Copyright 2011-2024, The Trustees of Indiana University and Northwestern
+#   University.  Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+# 
+# You may obtain a copy of the License at
+# 
+# http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software distributed
+#   under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+#   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+#   specific language governing permissions and limitations under the License.
+# ---  END LICENSE_HEADER BLOCK  ---
+
+module Avalon
+  class FFprobe
+    # media_path should be the output of `FileLocator.new(file_location).location`
+    def initialize(media_path)
+      ffprobe = Settings&.ffprobe&.path || 'ffprobe'
+      raw_output = `#{ffprobe} -i "#{media_path}" -v quiet -show_format -show_streams -count_packets -of json`
+      # $? is a variable for the exit status of the last executed process. 
+      # Success == 0, any other value means the command failed in some way.
+      unless $?.exitstatus == 0
+        Rails.logger.error "File processing failed. Please ensure that FFprobe is installed and that the correct path is configured."
+        return
```

Comment:
What do you think about raising an exception here (maybe a generic `RuntimeError`) or setting `@json_output = {}` before returning (looks like the ActiveStorage analyzer does this)?  I think returning nil as it is right now will cause a NoMethodError in `MasterFile#reloadTechnicalMetadata`.